dcos_launch/aws.py

```diff
@@ -87,7 +87,9 @@ def delete(self):
         if len(self.config['temp_resources']) > 0:
             # must wait for stack to be deleted before removing
             # network resources on which it depends
-            self.stack.wait_for_complete(transition_states=['DELETE_IN_PROGRESS'], end_states=['DELETE_COMPLETE'])
+            self.stack.wait_for_complete(
+                transition_states=['CREATE_COMPLETE', 'DELETE_IN_PROGRESS'],
```

Comment:
You should consider adding a comment on why `CREATE_COMPLETE` is added here. The previous state values `DELETE_IN_PROGRESS` and `DELETE_COMPLETE` are intuitive.

The addition of `CREATE_COMPLETE` is not at all intuitive for this method.

Reply:
And why do we wait on the state-transitions at all? Why can't `try: delete-all: catch: pass` after the calling aws.stack.delete?

Reply:
We need to wait on the state transition so we can have the ability to delete resources (like AWS keypairs) that exist before the stack was created and therefore need to be deleted after its gone